singlePost.php

```diff
@@ -1,31 +1,67 @@
 <?php
+declare(strict_types=1);
+session_start();
+require_once 'src/services/PostDisplayService.php';
+require_once 'src/services/DatabaseConnectionService.php';
+require_once 'src/entities/PostEntity.php';
+require_once 'src/models/PostsModel.php';
+require_once 'src/entities/CommentEntity.php';
+require_once 'src/services/CommentValidationService.php';
+require_once 'src/models/CommentsModel.php';
 
- declare(strict_types=1);
- require_once 'src/services/PostDisplayService.php';
- require_once 'src/services/DatabaseConnectionService.php';
- require_once 'src/entities/PostEntity.php';
- require_once 'src/models/PostsModel.php';
+include_once 'header.php';
 
- $db = DatabaseConnectionService::connect();
- $PostsModel = new PostsModel($db);
+$db = DatabaseConnectionService::connect();
+$PostsModel = new PostsModel($db);
+$hideComments = false;
 
-$post = $PostsModel->getSingle((int)$_GET['id']);
+$contentError = "";
+$content = $_POST['content'] ?? '';
+$id = (int)$_GET['id'];
 
-if (!$post){
+$post = $PostsModel->getSingle($id);
+
+if (!$post) {
     header('Location: index.php');
 }
 
- require_once 'header.php';
-?>
+if ($_SERVER["REQUEST_METHOD"] === "POST"){
+    $contentValid = CommentValidationService::ContentValidation($content);
+    if (!$contentValid) {
+        $contentError = 'Content must be between 10 and 200 characters.';
+    }
 
+    if ($contentValid) {
+        $CommentModel = new CommentsModel($db);
+        $commentEntity = CommentsModel::AddCommentToDatabase($id, $content);
+        $CommentModel->addComment($commentEntity);
+        $hideComments = true;
+        $content = '';
+    }
+ }
 
-<section class="container md:w-1/2 mx-auto">
-    <article class="p-8 border border-solid rounded-md">
-        <?php
-        echo PostDisplayService::displaySingle($post);
+$comment = $PostsModel->getComments($id);
+echo PostDisplayService::displaySingle($post, $comment);
+
+if (!isset($_SESSION['loggedIn'])) {
+    exit();
```

Comment:
Do you really want to `exit()` if the user isn't logged in? The user will get an incomplete page.